tools/generate-bolts.py

```diff
@@ -529,31 +529,34 @@ def main(options, args=None, output=sys.stdout, lines=None):
                     raise ValueError('Unknown message type {}. {}:{}'.format(tokens[1], ln, line))
                 type_obj, collapse, optional = master.add_type(tokens[3], tokens[2])
 
-                # if this is an 'extension' field*, we want to add a new 'message' type
-                # in the future, extensions will be handled as TLV's
-                #
-                # *(in the spec they're called 'optional', but that term is overloaded
-                #   in that internal wire messages have 'optional' fields that are treated
-                #   differently. for the sake of clarity here, for bolt-wire messages,
-                #   we'll refer to 'optional' message fields as 'extensions')
-                #
-                if bool(tokens[5:]):  # is an extension field
-                    extension_name = "{}_{}".format(tokens[1], tokens[5])
-                    orig_msg = msg
-                    msg = master.find_message(extension_name)
-                    if not msg:
-                        msg = copy.deepcopy(orig_msg)
-                        msg.enumname = msg.name
-                        msg.name = extension_name
-                        master.add_extension_msg(msg.name, msg)
-
                 if collapse:
                     count = 1
                 else:
                     count = tokens[4]
 
-                msg.add_data_field(tokens[2], type_obj, count, comments=list(comment_set),
-                                   optional=optional)
+                if tokens[5:] == []:
+                    msg.add_data_field(tokens[2], type_obj, count, comments=list(comment_set),
+                                       optional=optional)
+                else:
+                    # if this is an 'extension' field*, we want to add a new 'message' type
+                    # in the future, extensions will be handled as TLV's
+                    #
+                    # *(in the spec they're called 'optional', but that term is overloaded
+                    #   in that internal wire messages have 'optional' fields that are treated
+                    #   differently. for the sake of clarity here, for bolt-wire messages,
+                    #   we'll refer to 'optional' message fields as 'extensions')
+                    #
+                    orig_msg = msg
+                    for extension in tokens[5:]:
```

Comment:
i think i'm missing something about the expected tokens value here. in my understanding, the input data that we'd expect for these fields is only ever of size 1, eg:

    msgdata,channel_update,htlc_maximum_msat,u64,,option_channel_htlc_max

Reply:
i should really add one of these to `test_cases`

Reply:
The option_static_remotekey PR added a single (and probably our only ever!) case of this:

```diff
@@ -1117,7 +1129,7 @@ messages are), they are independent of requirements here.
    * [`channel_id`:`channel_id`]
    * [`u64`:`next_local_commitment_number`]
    * [`u64`:`next_remote_revocation_number`]
-   * [`32*byte`:`your_last_per_commitment_secret`] (option_data_loss_protect)
+   * [`32*byte`:`your_last_per_commitment_secret`] (option_data_loss_protect,option_static_remotekey)
    * [`point`:`my_current_per_commitment_point`] (option_data_loss_protect)
 
 `next_local_commitment_number`: A commitment number is a 48-bit
```